src/SdfEntityCreator.cc

```diff
@@ -788,8 +790,33 @@ Entity SdfEntityCreator::CreateEntities(const sdf::Visual *_visual)
   // \todo(louise) Populate with default material if undefined
   if (_visual->Material())
   {
+    sdf::Material visualMaterial = *_visual->Material();
+    if (!_visual->Material()->ScriptUri().empty())
+    {
+      std::string scriptUri = visualMaterial.ScriptUri();
+      if (scriptUri != "file://media/materials/scripts/gazebo.material") {
+        gzwarn << "Custom material scripts are not supported."
+          << std::endl;
+      }
+    }
+    if (!_visual->Material()->ScriptName().empty())
+    {
+      std::string scriptName = visualMaterial.ScriptName();
+
+      if((scriptName.find("Gazebo/") != std::string::npos)) {
+        gzwarn << "Using internal gazebo.material to parse " << scriptName
+          << std::endl;
+        MaterialParser* materialParser = new MaterialParser();
+        materialParser->Load();
+        std::vector<std::vector<float>> parsed = materialParser->GetMaterialValues(scriptName);
+
+        visualMaterial.SetAmbient(gz::math::Color(parsed[0][0],parsed[0][1], parsed[0][2]));
```

Comment:
This currently segfaults if the material doesn't have all three materials. If we return a struct from `GetMaterialValues`, we wouldn't have this problem.